src/main/java/com/fasterxml/jackson/databind/node/DoubleNode.java

```diff
@@ -52,6 +52,13 @@
     @Override
     public boolean isDouble() { return true; }
 
+    @Override
+    public boolean hasFractionalPart() {
+        long iPart = (long) _value;
+        double fPart = _value - iPart;
+        return fPart != 0.0;
```

Comment:
Not sure if this works reliably (or needs Epsilon diff check) -- would probably good to check; if you have a link to somewhere that says this works reliably that'd be good. Not sure if coercion to `long` could lose also not work in some cases as `double` magnitude extends beyond `long` domain.

Reply:
Good point thanks, let me investigate more on this.

Reply:
I think SO helps again: as per https://stackoverflow.com/questions/9898512/how-to-test-if-a-double-is-an-integer we could use `Math.rint()` like

```
if (x == Math.rint(x)) { // is integral
```

I am not 100% sure if NaN has to be checked separately; possibly not given that NaN typically is NOT the same is NaN (wrt Java `==`), but it might be fine to use full suggested comparison (from linked-to Guava impl https://guava.dev/releases/19.0/api/docs/com/google/common/math/DoubleMath.html#isMathematicalInteger(double))

```
!Double.isNaN(x) && !Double.isInfinite(x) && x == Math.rint(x)
```

... and since `DoubleNode` already defines `isNaN()` that does this, can just call that before or after using `Math.rint()`?

Reply:
I change this part to something very simple and reliable, what do you think about the new one?